plugins/grafana-custom-plugins/grafana-dependency-plugin/src/DependencyPanel.tsx

```diff
@@ -47,124 +49,165 @@ export const DependencyPanel: React.FC<Props> = ({ options, data, width, height
+      let dests = new Map<string, number>();
+      dests.set(pod_dst, octetDeltaCount);
+      if (destinationServicePortName !== '') {
+        dests.set(svc_dst, octetDeltaCount);
+      }
+      if (srcToDestMap.has(pod_src)) {
+        if (srcToDestMap.get(pod_src)?.has(pod_dst)) {
+          srcToDestMap.get(pod_src)?.set(pod_dst, octetDeltaCount+srcToDestMap.get(pod_src)?.get(pod_dst));
+        } else {
+          srcToDestMap.get(pod_src)?.set(pod_dst, octetDeltaCount);
+        }
+        if (destinationServicePortName === '') {
+          continue;
+        } else if (srcToDestMap.get(pod_src)?.has(svc_dst)) {
+          srcToDestMap.get(pod_src)?.set(svc_dst, octetDeltaCount+srcToDestMap.get(pod_src)?.get(svc_dst));
+        } else {
+          srcToDestMap.get(pod_src)?.set(svc_dst, octetDeltaCount);
+        }
+      } else {
+        srcToDestMap.set(pod_src, dests);
       }
-      return sourcePodName;
     }
+  
+    // format pods inside node within graph string
+    nodeToPodMap.forEach((pods, nodename) => {
+      let str = 'subgraph ' + nodename + '\n';
+      pods.forEach((pod) => {
+        str += nodename + '_pod_' + pod + '(' + pod + ');\n';
+      });
+      str += 'end;\n';
+      graphString += str;
+    });
+    // format arrows to services and pods within graph string
+    let prefixes = ['', 'K', 'M', 'G', 'T'];
+    srcToDestMap.forEach((destsToBytes, src) => {
+      destsToBytes.forEach((bytes, dest) => {
+        let usedpref = Math.floor(Math.log(bytes) / Math.log(1000));
+        if (usedpref > 4) {usedpref = 4};
+        let str = src + ' -- ' + bytes/(Math.pow(1000, usedpref)) + ' ' + prefixes[usedpref] + 'B --> ' + dest + ';\n';
+        graphString += str;
+      });
+    });
+  }
 
-    let groupByPodLabel = options.groupByPodLabel;
-    let srcName = getName(groupByPodLabel, true, sourcePodLabel);
-    let dstName = getName(groupByPodLabel, false, destinationPodLabel);
-
-    // determine which nodes contain which pods
-    if (nodeToPodMap.has(sourceNodeName) && !nodeToPodMap.get(sourceNodeName)?.includes(srcName)) {
-      nodeToPodMap.get(sourceNodeName)?.push(srcName);
-    } else if (!nodeToPodMap.has(sourceNodeName)) {
-      nodeToPodMap.set(sourceNodeName, [srcName]);
-    }
-    if (nodeToPodMap.has(destinationNodeName) && !nodeToPodMap.get(destinationNodeName)?.includes(dstName)) {
-      nodeToPodMap.get(destinationNodeName)?.push(dstName);
-    } else if (!nodeToPodMap.has(destinationNodeName)) {
-      nodeToPodMap.set(destinationNodeName, [dstName]);
+  function layerSevenGraph() {
+    function getColorFromStatus(httpStatus: string) {
+      // colors that correspond to each of the types of http response code; i.e. 4xx and 5xx codes return red to symbolize errors
+      const codeToColor = {1:'orange', 2:'green', 3:'blue', 4:'red', 5:'red'};
+      let statusType = +httpStatus.charAt(0);
+      if (statusType < 1 || statusType > 5) {
```

Comment:
Is it possible for us to get statusType < 1 or > 5? Is there such a http status code? I see you added it for indicating errors, but I wonder whether it is possible to happen, if it does happen, would it be better to catch this error in Flow Exporter or Flow Aggregator? cc @tushartathgur

Reply:
I don't think it should, but we can double check with @tushartathgur.